Programming Course Draft 4.R

```diff
@@ -0,0 +1,433 @@
+
+  }
+
+
+  # warning messages for incorrect data input
+
+  last_minute <- tail(data[,1])
+
+  last_minute <- last_minute[0]
+
+  if (last_minute < .5) {
+
+    warning("This software is not appropriate for heart rate
+            segments smaller than 30 seconds.")
+
+  }
+
+  if (segment = TRUE) {
+
+    if (min > max) {
+
+      warning("Incorrect min/max values. Max value should exceed min value by
+              at least 30 seconds.")
+
+    }
+
+    range <- max - min
+
+    if (max - min < 0.5) {
+
+      warning("This software is not appropriate for heart rate
+            segments smaller than 30 seconds.")
+
+    }
+
+  }
+
+
+  return(data)
+
+}
+
+
+
+#' Part Two: Time-Domain Statistical Methods
+
+#' Root Mean Squared Successive Differences (RMSSD)
+
+#' RMSSD is calculated using the successive differences ellapsed between heart
+#' beats (RR values). This is considered the most precise marker for parasympathetic
+#' activity using heart rate, and it is the most common time-domain HRV statistic reported
+#' in the literature.
+#'
+#' RMSSD values can be subject to individul differences, so it is best to compare RMSSD
+#' values between groups or across time, rather than a single value indicating a validated
+#' level of distress.
+#'
+#' Lower HRV and RMSSD values indicate higher levels of distress.
+#'
+#' @param infile Path to the input file
+#' @return A matrix of the infile
+#' @export
+
+RMSSD_Calculator  <- function(data = data) {
+
+    input_length <- length(data[,2]) - 1
+
+    RR_squared <- data^2
+
+    RR_squared_average <- sum(RR_squared) / input_length
+
+    rmssd1 <- sqrt(RR_squared_average)
+
+    return(rmssd1)
+
+}
+
+
+
+#' SDNN Function
+
+#' SDNN measurement is the standard deviation of all time ellapses heart beats.
+#' This value indicates both parasympathetic and sympathetic nervous system
+#' activity and is imporant measure in cardiology research and practice
+#' It is not often used in psychological research and is most often used in medical research
+#' when heart rate recordings are more than 24 hours. Values below 50ms indicate risk,
+#' while values above 100ms are relativily healthy
+#'
+#' @param infile Path to the input file
+#' @return A matrix of the infile
+#' @export
+
+
+SDNN_Calculation <- function(data) {
+
+  last_minute <- tail(data[,1])
+
+  last_minute <- last_minute[5]
+
+  if (last_minute < 1440) {
+
+    warning("SDNN is most accurate with heart rate segments longer than 24 hours.")
+
+  }
+
+  SDNN <- sd(data[,2])
+
+  return(SDNN)
+
+}
+
+
+#' NN50 Calculation
+
+#' NN50 number of intervals between heart beats that are greater than 50ms.
+#' This measure is used more in medical research and practice than psychological research.
+#' This is more useful when calculated as a percentage (next function).
+#'
+#' @param infile Path to the input file
+#' @return A matrix of the infile
+#' @export
+
+
+NN50_Count_Function <- function(data = data) {
```

Comment:
I might be wrong, but this looks like, depending on the data, to be doable outside an if-loop: 
counter <- nrow(data[ ,2]) - 50
if( counter < 1) { 
counter <- 0
}